crates/nu-parser/src/parse_keywords.rs

```diff
@@ -4006,52 +4020,57 @@ pub fn parse_register(
 ///   b) current working directory (PWD)
 ///
-/// Then, if the file is not found in the actual cwd, NU_LIB_DIRS is checked.
-/// If there is a relative path in NU_LIB_DIRS, it is assumed to be relative to the actual cwd
+/// Then, if the file is not found in the actual cwd, dirs_var is checked.
+/// If dirs_var is an Expr::Var, then we look for a const with that VarId,
+/// and if dirs_var is an Expr::String, then we look for an environment with that name.
+/// If there is a relative path in dirs_var, it is assumed to be relative to the actual cwd
 /// determined in the first step.
 ///
 /// Always returns an absolute path
 pub fn find_in_dirs(
     filename: &str,
     working_set: &StateWorkingSet,
     cwd: &str,
-    dirs_env: &str,
+    dirs_var: Option<&Expr>,
 ) -> Option<PathBuf> {
     // Choose whether to use file-relative or PWD-relative path
     let actual_cwd = if let Some(currently_parsed_cwd) = &working_set.currently_parsed_cwd {
         currently_parsed_cwd.as_path()
     } else {
         Path::new(cwd)
     };
-
     if let Ok(p) = canonicalize_with(filename, actual_cwd) {
-        Some(p)
-    } else {
-        let path = Path::new(filename);
-
-        if path.is_relative() {
-            if let Some(lib_dirs) = working_set.get_env_var(dirs_env) {
-                if let Ok(dirs) = lib_dirs.as_list() {
-                    for lib_dir in dirs {
-                        if let Ok(dir) = lib_dir.as_path() {
-                            // make sure the dir is absolute path
-                            if let Ok(dir_abs) = canonicalize_with(dir, actual_cwd) {
-                                if let Ok(path) = canonicalize_with(filename, dir_abs) {
-                                    return Some(path);
-                                }
-                            }
-                        }
-                    }
+        return Some(p);
+    }
 
-                    None
-                } else {
-                    None
-                }
-            } else {
-                None
-            }
-        } else {
-            None
-        }
+    let path = Path::new(filename);
+    if !path.is_relative() {
+        return None;
     }
+
+    let lib_dirs = match dirs_var {
+        Some(Expr::Var(var_id)) => working_set.find_constant(*var_id),
+        Some(Expr::String(dirs_env_var)) => working_set.get_env_var(dirs_env_var),
+        _ => return None,
+    }?;
+
+    lib_dirs
+        .as_list()
+        .ok()?
+        .iter()
+        .map(|lib_dir| -> Option<PathBuf> {
+            let dir = lib_dir.as_path().ok()?;
+            let dir_abs = canonicalize_with(dir, actual_cwd).ok()?;
+            canonicalize_with(filename, dir_abs).ok()
+        })
+        .find(Option::is_some)
+        .flatten()
+}
+
+pub fn find_dirs_var(working_set: &StateWorkingSet, var_name: &str) -> Expr {
+    working_set
+        .find_variable(format!("${}", var_name).as_bytes())
```

Comment:
I don't think you need to prepend the `$`. Does just `find_variable(var_name.as_bytes())` work?

Reply:
Unfortunately not. I had it without prepending $ at first and it didn't work until I added it.

Reply:
I'm guessing that this is because if you do `$nu.scope.var` all the variables start with `$`. Another way to fix this is to take the `$` off when they're being created and stored. I never really understood why we were artificially prepending a `$` in the first place, but there could be a valid reason.

You can see where we add it here
https://github.com/nushell/nushell/blob/26c36e932e62728b7930643c2f1b3ef0e906ae36/crates/nu-protocol/src/engine/engine_state.rs#L1807-L1810

This would probably have to change too
https://github.com/nushell/nushell/blob/26c36e932e62728b7930643c2f1b3ef0e906ae36/crates/nu-parser/src/parser.rs#L117-L123

Reply:
I see. Yeah, seems unnecessary to keep adding the `$` and can probably be removed, but also it's quite off-topic for this PR so it's not worth spending too much time on. It could be fixed in another PR.

Reply:
Agreed. Just wanted to bring it up since I found it recently doing other `$` stuff.

Reply:
yes, please fix that. I originally had it in there for debugging purposes and forgot to fix it.